frontend/src/utils/helpers/moduleUtils.js

```diff
@@ -0,0 +1,38 @@
+/* eslint-disable no-restricted-properties */
+import config from '../../generated/config.json';
+
+const ModuleNames = {
+  CATALOG: 'catalog',
+  DATASETS: 'datasets',
+  SHARES: 'shares',
+  GLOSSARIES: 'glossaries',
+  WORKSHEETS: 'worksheets',
+  NOTEBOOKS: 'notebooks',
+  MLSTUDIO: 'mlstudio',
+  PIPELINES: 'datapipelines',
+  DASHBOARDS: 'dashboards'
+};
+
+function isModuleEnabled(module) {
+  if (module === ModuleNames.CATALOG) {
```

Comment:
This means Catalog can only be enabled if Datasets or Dashboards is enabled, is this the only module that have this conditional dependency? I think, if this can occur for other modules, it will be better to just create a method that handles that generally. That way it is reusable.

Reply:
This function acts as the single handler for checking module enablement. You can adjust the rules depending on your module's requirements. It acts as a `switch` case for the modules. In our case, only `catalog` has the a complex condition to be enabled, and the default behavior for other modules, is simply checking the `config.json` for `active` status. For future modules, you can adjust the condition by simply inserting an `if` case to check your module, and add your checks accordingly.

Reply:
I think could make it more flexible allowing to pass a list of module names and do checks on each to see if module should be loaded ... but for now this PR looks good to me and can think on if something like that would help moving forward